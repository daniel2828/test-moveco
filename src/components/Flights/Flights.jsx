import React from 'react';
import Flight from '../Flight';
import BpkCard from 'bpk-component-card';
class Flights extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      jsonLoaded: false,
      itineraries: [],
      legs: [],
      error: null,
    };
  }
  componentDidMount() {
    fetch('flights.json')
      .then(res => res.json())
      .then(
        result => {
          this.setState({
            jsonLoaded: true,
            itineraries: result.itineraries,
            legs: result.legs,
          });
        },

        error => {
          this.setState({
            isLoaded: true,
            error,
          });
        },
      );
  }
  getLegs = (item, legs) => {
    let itemsLegs = [];
    legs.map(leg => {
      if (leg.id === item.legs[0] || leg.id === item.legs[1]) {
        itemsLegs.push(leg);
      }
    });
    return itemsLegs;
  };
  render() {
    const { jsonLoaded, itineraries, legs, error } = this.state;
    if (error) {
      return <div>Error: {error.message}</div>;
    } else if (!jsonLoaded) {
      return <div>Loading...</div>;
    } else {
      return (
        <div className="flights-body">
          {itineraries.map(item => (
            <Flight
              key={item.id}
              itinerarie={item}
              legs={this.getLegs(item, legs)}
            />
          ))}
        </div>
      );
    }
  }
}
export default Flights;
