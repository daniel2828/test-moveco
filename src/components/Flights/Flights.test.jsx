import React from 'react';
import TestRenderer from 'react-test-renderer';

import Flights from './Flights';

describe('Flights', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(<Flights />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
