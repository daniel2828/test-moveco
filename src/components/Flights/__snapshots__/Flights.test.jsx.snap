// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flights should render correctly 1`] = `
<div>
  Loading...
</div>
`;
